### PR TITLE
Fix #1142: Pagination Support

### LINKED ISSUE #1142: Pagination support
According to [Kubernetes docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#list-deployment-v1-apps), most, if not all list operations support pagnitation through query parameters `limit` and `continue`. But currently the client doesn't seem to have support for it. 

Do you have plans to provide pagination for list operations in the near future?

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fixes #1142

### CODE REVIEW

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ListExamples.java [hunk 1]
```diff
@@ -55,6 +57,12 @@ public static void main(String[] args) {
       System.out.println(
         client.pods().inNamespace("test").withName("testing").get()
       );
+      PodList podList = client.pods().inNamespace("myproject").list(10, "eyJ2IjoibWV0YS5rOHMuaW8vdjEiLCJydiI6OTIyNDE5LCJzdGFydCI6ImZhYnJpYzgtbWF2ZW4tc2FtcGxlLWV4dGVybmFsLXJlc291cmNlcy1zMmktNC1idWlsZFx1MDAwMCJ9");
```
Comment (hrishin) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ListExamples.java]: May like elaborate what is `continue` parameter?

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ListExamples.java [hunk 2]
```diff
@@ -55,6 +57,12 @@ public static void main(String[] args) {
...
+      for(Pod pod : podList.getItems()) {
```
Comment (hrishin) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ListExamples.java]: Can use lambda?

```
podList.getItems()
.stream()
.forEach(System.out::println);
```

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java [hunk 3]
```diff
@@ -171,6 +171,35 @@ private static String trimVersion(String apiVersion) {
...
+   *
+   * @param url
+   * @return list of corresponding Kubernetes Resources
+   */
+  private L listRequestHelper(URL url) {
+    try {
+      HttpUrl.Builder requestUrlBuilder = HttpUrl.get(url).newBuilder();
+
+      String labelQueryParam = getLabelQueryParam();
+      if (Utils.isNotNullOrEmpty(labelQueryParam)) {
```
Comment (hrishin) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]:
```java
 if (Utils.isNotNullOrEmpty(labelQueryParam)) {
    requestUrlBuilder.addQueryParameter("labelSelector", labelQueryParam);
 }
```

Could extract in method like

```java
void addQSParam(String name, value) {

}
```

then call it
```java
addQSParam("labelSelector", labelQueryParam);
addQSParam("fieldSelector", fieldQueryString);
```

DRY (Dont repeat yourself)

**Comment by hrishin** [comment]
@rohanKanojia, thanks for addressing comments!

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**